File: frontend/src/types/documents.ts
export type DocumentStatus = 'processing' | 'completed' | 'failed'

export interface DocumentListItemDto {
  id: string
  name: string
  status: DocumentStatus
  chunk_count: number
  uploaded_by: string
  created_at: string
  updated_at: string
}

export interface UploadedDocumentDto {
  id: string
  name: string
  status: DocumentStatus
  chunk_count: number
  uploaded_by: string
  created_at: string
  updated_at: string
}

export interface DocumentDetailDto {
  id: string
  name: string
  status: DocumentStatus
  chunk_count: number
  uploaded_by: string
  created_at: string
  updated_at: string
}

export interface DocumentListRequest {
  keyword?: string
  status?: DocumentStatus
  page?: number
  page_size?: number
}

export interface UploadDocumentRequest {
  file: File
  name?: string
}

export interface DocumentListResponseData {
  items: DocumentListItemDto[]
  total: number
  page: number
  page_size: number
}

export interface UploadDocumentResponseData {
  document: UploadedDocumentDto
}

export interface DocumentDetailResponseData {
  document: DocumentDetailDto
}

export interface DeleteDocumentResponseData {
  deleted: boolean
  document_id: string
}

File: backend/src/models/documents.py
"""Document ingestion API DTOs aligned with docs/api-contracts.md."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

DocumentStatus = Literal["processing", "completed", "failed"]


class DocumentRead(BaseModel):
    id: str
    name: str
    status: DocumentStatus
    chunk_count: int
    uploaded_by: str
    created_at: str
    updated_at: str


class DocumentListResponseData(BaseModel):
    items: list[DocumentRead]
    total: int
    page: int
    page_size: int


class DocumentMutationResponseData(BaseModel):
    document: DocumentRead


class DocumentDetailResponseData(BaseModel):
    document: DocumentRead


class DocumentDeleteResponseData(BaseModel):
    deleted: bool
    document_id: str
